### PR TITLE
Add WebTransport

### PR DESCRIPTION
**Pull request by osdodo**

### CODE REVIEW

File: src/helper.ts [hunk 1]
```diff
@@ -74,3 +85,35 @@ export function updateQueryStringParameter(
...
+/**
+ * Function for obtaining authorised URL
+ *
+ * @param host Service url
+ * @param {IPresenceOption} option
+ *
+ * @returns Promise containing AuthorizedURL
+ */
+export async function getAuthorizedURL(host: string, option: IPresenceOption) {
+    if (option?.auth?.type === 'publickey' && option.auth.publicKey) {
```
Comment (fanweixiao) [inline src/helper.ts:98]: @osdodo @xiaojian-hong as we guarantee security-first from dev to test to production, `publickey` based authentication is not secure for production, shall we disable `publickey` here in the future?

File: src/index.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
 import Presence from './presence';
 
-export { Presence };
+export default Presence;
```
Comment (fanweixiao) [inline src/index.ts:3]: Good for developers !

File: src/websocket.ts [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+    /**
+     * Connect to YoMo
+     *
+     * @private
+     */
+    private async _connect() {
+        if (!this._wasmLoaded) {
+            try {
+                await loadWasm('https://d1lxb757x1h2rw.cloudfront.net/y3.wasm');
```
Comment (fanweixiao) [inline src/websocket.ts:163]: Suggest: use a const for this url

File: src/webtransport.ts [hunk 4]
```diff
@@ -0,0 +1,218 @@
+import { interval, Observable, Subject, Subscription } from 'rxjs';
+import { distinctUntilChanged, filter, map } from 'rxjs/operators';
+import { decoder, encoder, getAuthorizedURL, loadWasm } from './helper';
+import { EventMessage, IPresence, IPresenceOption } from './type';
+
+export default class WT extends Subject<EventMessage> {
```
Comment (fanweixiao) [inline src/webtransport.ts:6]: Shall we introduce an abstract for `Transport` as base class or interface for `WT` and `WS` in the future?

**Review by fanweixiao — APPROVED** [review]

File: src/helper.ts (hunk 1 above)
Comment (xiaojian-hong) [reply src/helper.ts:98]: yes, I agree.

File: README.md [hunk 5]
```diff
@@ -118,10 +119,10 @@ Response data:
 #### Create a `Presence` instance.
 
 ```js
-import { Presence } from '@yomo/presencejs';
+import Presence from '@yomo/presencejs';
 
 // create an instance.
-const yomo = new Presence('wss://presence.yomo.dev', {
+const yomo = new Presence('https://presence.yomo.dev', {
```
Comment (xiaojian-hong) [inline README.md:125]: I think we can replace the URL `https://presence.yomo.dev'` with `https://prsc.yomo.dev`, and introduce how to get the test/dev account on https://presencejs.yomo.run? like we did for `react-cursor-chat`.
https://github.com/yomorun/react-cursor-chat#request-free-devtest-account

**Review by fanweixiao — APPROVED** [review]

**Review by xiaojian-hong — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (fanweixiao) [reply README.md:125]: oh yes, we have to do this.

**Pull request merged**